packages/salesforcedx-vscode-core/test/vscode-integration/providers/configAggregatorProvider.test.ts

```diff
@@ -0,0 +1,205 @@
+      // Arrange
+      getCurrentDirectoryStub.returns(dummyProjectRootWorkspacePath);
+      getRootWorkspacePathStub.returns(dummyProjectRootWorkspacePath);
+
+      // Act
+      const configAggregator = await (configAggregatorProvider as any).createConfigAggregator();
+
+      // Assert
+      // createConfigAggregator should store the cwd initially,
+      // and check it again after creating the ConfigAggregator
+      // to ensure that the cwd is set back to its original value.
+      expect(getCurrentDirectoryStub.callCount).to.equal(2);
+      // Since the stubbed current directory is the same as the root
+      // workspace path, the directory should not have been changed.
+      expect(changeCurrentDirectoryToStub.callCount).to.equal(0);
+      expect(configAggregatorCreateSpy.callCount).to.equal(1);
+      expect(configAggregator).to.not.equal(undefined);
+      expect(getRootWorkspacePathStub.callCount).to.equal(1);
+    });
+
+    it('should create an SfdxConfigAggregator', async () => {
+      // Arrange
+      getCurrentDirectoryStub
+        .onCall(0)
+        .returns(ConfigAggregatorProvider.defaultBaseProcessDirectoryInVSCE);
+      const sfdxConfigAggregatorCreateSpy = sandbox.spy(
+        SfdxConfigAggregator,
+        'create'
+      );
+
+      // Act
+      const sfdxConfigAggregator = await (configAggregatorProvider as any).createConfigAggregator(
+        { sfdx: true }
+      );
+
+      // Assert
+      expect(sfdxConfigAggregatorCreateSpy.callCount).to.equal(1);
+      expect(sfdxConfigAggregator).to.not.equal(undefined);
+    });
+
+    it('should change back to the original current directory if ConfigAggregator creation fails', async () => {
+      // Arrange
+      configAggregatorCreateSpy.restore();
+      const configAggregatorCreateStub = Sinon.stub(ConfigAggregator, 'create');
+      configAggregatorCreateStub.throws(
+        new Error('There was a problem creating the Config Aggregator.')
+      );
+      getCurrentDirectoryStub
+        .onCall(0)
+        .returns(ConfigAggregatorProvider.defaultBaseProcessDirectoryInVSCE);
+
+      // Act
+      let configAggregator;
+      let caughtError;
+      try {
+        configAggregator = await (configAggregatorProvider as any).createConfigAggregator();
+      } catch (error) {
+        caughtError = error;
+      }
+
+      // Assert
+      // createConfigAggregator should store the cwd initially,
+      // and check it again after creating the ConfigAggregator
+      // to ensure that the cwd is set back to its original value.
+      expect(getCurrentDirectoryStub.callCount).to.equal(2);
+      expect(changeCurrentDirectoryToStub.callCount).to.equal(2);
+      expect(changeCurrentDirectoryToStub.getCall(1).args[0]).to.equal(
+        ConfigAggregatorProvider.defaultBaseProcessDirectoryInVSCE
+      );
+      expect(configAggregatorCreateStub.callCount).to.equal(1);
+      expect(caughtError).to.not.equal(undefined);
```

Comment:
nit: typically it's good to verify your error here to ensure the thrown error was the one you expected. Typically I'd make 'There was a problem creating the Config Aggregator.' a const and pass that to the new Error and use it in the expect ala `expect(error.message).to.equal(errMessage);`